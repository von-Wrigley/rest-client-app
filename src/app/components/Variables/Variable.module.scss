@use "../../../styles/variables.scss" as *;
@use "../../../styles/mixins.scss" as *;

.variablesWrapper {
}

.addButton {
  @include button-style;
  padding: 12px 40px;
  margin-bottom: 10px;
}

.inputWrapper {
  display: flex;
  gap: 8px;
  margin-bottom: 10px;
}

.input {
  @include input-style;
}

.button {
  @include button-style;
}
