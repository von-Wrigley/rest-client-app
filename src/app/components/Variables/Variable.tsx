"use client";

import { useLocalStorage } from "@/app/hooks/LocStor";
import React, { useEffect, useState } from "react";
import styles from "./Variable.module.scss";

type Variable = {
  key: string;
  value: string;
};

function VariablesComponent() {
  const [variablestorage, setVariableStorage] = useLocalStorage(
    "variables",
    [],
  );

  const [inputs, setInputs] = useState([{ key: "", value: "" }]);

  useEffect(() => {
    setInputs(variablestorage);
  }, [variablestorage]);

  const handleAddVariable = () => {
    setInputs([...inputs, { key: "", value: "" }]);
  };

  const handleRemoveVariable = (index: number) => {
    const newArray = [...inputs];
    newArray.splice(index, 1);
    setInputs(newArray);
  };

  const handleVariable = (
    e: React.ChangeEvent<HTMLInputElement>,
    index: number,
  ) => {
    const { name, value } = e.target;
    const list = [...inputs];
    list[index][name] = value;
    setInputs(list);

    if (typeof window !== "undefined") {
      setVariableStorage(inputs);
    }
  };

  return (
    <div className="border flex flex-col gap-y-3 p-2.5  ">
      <button
        onClick={handleAddVariable}
        type="button"
        className="bg-green-400 w-18 py-1.5"
      >
        Add
      </button>

      {inputs?.map((item: Variable, index: number) => (
        <div className="flex gap-x-2 " key={index}>
          <input
            type="text"
            placeholder="key"
            name="key"
            value={item.key}
            onChange={(e) => handleVariable(e, index)}
            className="border border-gray-300 "
          />
          <input
            type="text"
            placeholder="value"
            name="value"
            value={item.value}
            onChange={(e) => handleVariable(e, index)}
            className="border border-gray-300 "
          />
          <button
            type="button"
            onClick={() => handleRemoveVariable(index)}
            className="bg-red-600 p-3"
          >
            Delete
          </button>
        </div>
      ))}
    </div>
  );
}

export default VariablesComponent;
