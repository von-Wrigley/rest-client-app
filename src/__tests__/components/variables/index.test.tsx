import { render, screen } from '@testing-library/react'; 
import { useTranslations } from 'next-intl';
import VariablesComponent from '@/app/components/Variables/Variable';
import Variables from '@/components/variables';

jest.mock('next-intl', () => ({
  useTranslations: jest.fn(),
}));

jest.mock('@/app/components/Variables/Variable', () => 
  jest.fn(() => <div data-testid="variables-component" />)
);

describe('Variables Page Component', () => {
  const mockT = jest.fn((key) => key);
  
  beforeEach(() => {
    (useTranslations as jest.Mock).mockReturnValue(mockT);
    mockT.mockClear();
  });

  it('should render without crashing', () => {
    render(<Variables />);
    expect(screen.getByTestId('variables-component')).toBeInTheDocument();
  });

  it('should apply the correct header class', () => {
    render(<Variables />);
    const header = screen.getByRole('heading', { level: 1 });
    expect(header).toHaveClass('header');
  });

  it('should render the VariablesComponent', () => {
    render(<Variables />);
    expect(VariablesComponent).toHaveBeenCalled();
    expect(screen.getByTestId('variables-component')).toBeInTheDocument();
  });

  it('should match the expected structure', () => {
    const { container } = render(<Variables />);
    expect(container.firstChild).toMatchSnapshot();
  });
});