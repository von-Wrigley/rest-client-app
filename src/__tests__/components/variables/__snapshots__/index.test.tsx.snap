// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Variables Page Component should match the expected structure 1`] = `
<div>
  <div>
    <h1
      class="header"
    >
      variables
    </h1>
    <div
      data-testid="variables-component"
    />
  </div>
</div>
`;
